File: lib/features/order/domain/entities/order.dart
class OrderEntity {
  final String id;
  final String restoName;
  final String restoImageUrl;
  final List<String> items;
  final int total;
  final String status;
  final String date;

  const OrderEntity({
    required this.id,
    required this.restoName,
    required this.restoImageUrl,
    required this.items,
    required this.total,
    required this.status,
    required this.date,
  });
}

File: backend/internal/models/order.go
package models

type Order struct {
	ID           string   `json:"id"`
	RestaurantID string   `json:"restaurant_id"`
	RestoName    string   `json:"resto_name"`
	RestoImage   string   `json:"resto_image"`
	Items        []string `json:"items"`
	Total        int      `json:"total"`
	Status       string   `json:"status"`
	Date         string   `json:"date"`
}

type OrderResponse struct {
	Success bool    `json:"success"`
	Message string  `json:"message"`
	Data    []Order `json:"data"`
}

type TrackingStatus struct {
	OrderID string `json:"order_id"`
	Status  string `json:"status"`
	Message string `json:"message"`
	Step    int    `json:"step"`
}